src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -177,14 +178,32 @@ public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener lis
         return true;
     }
 
+    private static String getBuildStepName(BuildStep s) {
+        if (s instanceof Describable) {
+            return String.format("%s (%s)", ((Describable<?>)s).getDescriptor().getDisplayName(), s.toString());
+        } else {
+            return s.toString();
+        }
+    }
+
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
+        // do as AbstractBuild.AbstractRunner#performAllBuildSteps
+        boolean wholeResult = true;
         for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
-                return false;
```

Comment:
If runner return false, then next runner/actions wouldn't run at all. This is good for performance for not doing some console string search every time.
ConditionalPublisher - just do grouping of actions and it should execute evaluation every time to ensure that condition is right, "evaluate on failure" controls returned boolean and build status.